handbook/direction/code-graph/code-insights/index.md

```diff
@@ -1,44 +1,218 @@
+
+As Sourcegraph becomes an increasingly powerful developer tool, Code Insights will be vital to tracking and understanding how both Sourcegraph and a company's engineers impact a codebase.
+
+**What this looks like:**
+
+- Code Insights will be extensible and integrate with other Sourcegraph features
+
+**What this does _not_ look like:**
+
+- We won't add integrations just for the sake of integrations, without a validated need and use case
+
+### Code Insights is NOT for "tracking individual developers"
+
+Code Insights is not built for trying to track an individual's output and whether or not someone is a "good" engineer or "high performing" employee. There are (unfortunately) other tools that exist that have made the mistake of tracking easy-to-find, pointless-to-measure metrics like "number of commits" or "number of lines of code" an individual contributes.
+
+**What this looks like:**
+
+- We don't build features that would primarily be used for evaluating individuals' performance
+- On a case by case basis, we might even intentionally build features to _block_ our other features from being used for this purpose, if the need ever arises
+- We won't market or sell code insights under false pretenses that this is a valid use case for this feature
+
+**What this does _not_ look like:**
+
+- We won't prevent the positive use cases of "individual tracking": if an individual wants to make a code insight demonstrating – for example – the spread of an API they built, they can
+
+### Code Insights is an enterprise quality professional tool
+
+For Code Insights to be valuable, it must be precise, exact, and performant. When it comes to codebases, there's little value in a measurement tool that is "90%" correct, or "within +/- 5%": our users need exact answers to inform decisions and track goals successfully. Code Insights also has to be performant enough that one can use it to answer questions when the need arises – our users can't wait weeks for their data, nor can they "ration" use of Code Insights among a Sourcegraph instance.
+
+**What this looks like:**
+
+- We don't compromise on precise accuracy; we test thoroughly, often, and at the extremes of possible use
+- We often prioritize improvements to speed and database size above more user-facing tweaks
+
+**What this does _not_ look like:**
+
+- Not every Code Insights feature will be "instant" – our developer audience understands the complexity of the tool and forgives some amount of loading/processing time
+
+### Code Insights builds on top of Sourcegraph's other features
+
+Users do not need to learn a new query language or format to use Code Insights. Nor will Code Insights introduce features that function differently than those "same" features in other areas of the product, confusing users. Code Insights is built atop Sourcegraph Search and will build with Sourcegraph's other features rather than implement its own versions of any functionalit.
```

Comment:
```suggestion
Users do not need to learn a new query language or format to use Code Insights. Nor will Code Insights introduce features that function differently than those "same" features in other areas of the product, confusing users. Code Insights is built atop Sourcegraph Search and will build with Sourcegraph's other features rather than implement its own versions of any functionality.
```